Sources/SwiftBSON/BSONRegularExpression.swift

```diff
@@ -81,22 +82,36 @@ extension BSONRegularExpression: BSONValue {
      *   - `DecodingError` if `json` is a partial match or is malformed.
      */
     internal init?(fromExtJSON json: JSON, keyPath: [String]) throws {
-        // canonical and relaxed extended JSON
-        guard let value = try json.value.unwrapObject(withKey: "$regularExpression", keyPath: keyPath) else {
-            return nil
-        }
-        guard
-            let (pattern, options) = try value.unwrapObject(withKeys: "pattern", "options", keyPath: keyPath),
-            let patternStr = pattern.stringValue,
-            let optionsStr = options.stringValue
-        else {
-            throw DecodingError._extendedJSONError(
-                keyPath: keyPath,
-                debugDescription: "Could not parse `BSONRegularExpression` from \"\(value)\", " +
-                    "\"pattern\" and \"options\" must be strings"
-            )
+        // canonical and relaxed extended JSON v2
+        if let regex = try json.value.unwrapObject(withKey: "$regularExpression", keyPath: keyPath) {
+            guard
+                let (pattern, options) = try regex.unwrapObject(withKeys: "pattern", "options", keyPath: keyPath),
+                let patternStr = pattern.stringValue,
+                let optionsStr = options.stringValue
+            else {
+                throw DecodingError._extendedJSONError(
+                    keyPath: keyPath,
+                    debugDescription: "Could not parse `BSONRegularExpression` from \"\(regex)\", " +
+                        "\"pattern\" and \"options\" must be strings"
+                )
+            }
+            self = BSONRegularExpression(pattern: patternStr, options: optionsStr)
+            return
+        } else {
+            // legacy / v1 extended JSON
```

Comment:
this parses the following format:
```
{
    "$regex": <string>,
    "$options": <string>,
}
```
which is unfortunately, the exact same format as a [query operator](https://docs.mongodb.com/manual/reference/operator/query/regex/).